### ISSUE TITLE
Custom Gradient compatible with Hybridize

### ISSUE THREAD

**Issue by xidulu** [apache/incubator-mxnet#17118]
## Description
Custom Backward is a feature supported by both tensorflow(https://www.tensorflow.org/api_docs/python/tf/custom_gradient?version=stable) and pytorch(https://pytorch.org/tutorials/beginner/examples_autograd/two_layer_net_custom_function.html). MXNet implements `autograd.Function` https://github.com/apache/incubator-mxnet/pull/7403/files , which is quite similar to the one in pytorch, however, it does not support either Symbolic mode or Hybridize
